Gems/Atom/RHI/Code/Include/Atom/RHI/MultiDeviceShaderResourceGroupData.h

```diff
@@ -181,6 +202,13 @@ namespace AZ::RHI
 
         ConstPtr<ShaderResourceGroupLayout> m_shaderResourceGroupLayout;
 
+        //! The backing data store of constants used only for the getters, actual storage happens in the single device SRGs.
+        ConstantsData m_constantsData;
```

Comment:
Curious as to why the MD version also needs to store a copy of ConstantsData?

Reply:
It has to do with the current discussion in discord. We removed it in this PR for now, since it is not necessary here yet. Let us continue the discussion on discord!

Reply:
Some more info on why `ConstantsData` was specifically introduced here:

- Most importantly (since it is not a test): `CubeMapRenderer.cpp`: `m_previousGlobalIblExposure = sceneSrg->GetConstant<float>(m_globalIblExposureConstantIndex.GetConstantIndex());` and the line below call the getters of the SRG.
- `MaterialTests.cpp` many times.
- `LuaMaterialFunctorTests.cpp` many times.
- And of course `ShaderResourceGroupConstantBufferTests.cpp` and the other RPI SRG Tests many times as well.

Reply:
Yes. If the user of this API is doing that, he should know why and what consequences that has. We don't have a way to prohibit that.